browser/brave_ads/brave_ads_host.cc

```diff
@@ -0,0 +1,162 @@
+/* Copyright (c) 2021 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "brave/browser/brave_ads/brave_ads_host.h"
+
+#include <memory>
+#include <string>
+#include <utility>
+
+#include "brave/browser/brave_ads/ads_service_factory.h"
+#include "brave/browser/brave_rewards/rewards_service_factory.h"
+#include "brave/browser/extensions/brave_component_loader.h"
+#include "brave/components/brave_ads/browser/ads_service.h"
+#include "brave/components/brave_rewards/browser/rewards_service.h"
+#include "chrome/browser/extensions/extension_service.h"
+#include "chrome/browser/profiles/profile.h"
+#include "chrome/browser/ui/browser.h"
+#include "chrome/browser/ui/browser_finder.h"
+#include "content/public/browser/web_contents.h"
+#include "extensions/browser/extension_system.h"
+#include "extensions/common/constants.h"
+
+namespace brave_ads {
+
+namespace {
+
+constexpr char kAdsEnableRelativeUrl[] = "request_ads_enabled_panel.html";
+
+}  // namespace
+
+BraveAdsHost::BraveAdsHost(content::WebContents* web_contents)
+    : WebContentsObserver(web_contents) {
+  DCHECK(web_contents);
+}
+
+BraveAdsHost::~BraveAdsHost() {
+  DCHECK(!callback_);
+}
+
+void BraveAdsHost::RequestAdsEnabled(RequestAdsEnabledCallback callback) {
+  DCHECK(callback);
+
+  Profile* profile =
+      Profile::FromBrowserContext(web_contents()->GetBrowserContext());
+  DCHECK(profile);
+
+  const AdsService* ads_service = AdsServiceFactory::GetForProfile(profile);
+  brave_rewards::RewardsService* rewards_service =
+      brave_rewards::RewardsServiceFactory::GetForProfile(profile);
+  if (!rewards_service || !ads_service || !ads_service->IsSupportedLocale()) {
```

Comment:
`IsSupportedLocale` doesn't seem like the right thing to do here, that should be internal to ads and covered by the pref value which is what I think you should be checking here

Reply:
cc @tmancey

Reply:
Maybe storing this in pref is not right because locale can be changed, but pref remains?
I think IsSupportedLocale should be checked here, because in case if we cannot turn on Ads we shouldn't show the popup.

Reply:
Anyway, if we need to make IsSupportedLocale I think we need to fix this

> `IsSupportedLocale` doesn't seem like the right thing to do here, that should be internal to ads and covered by the pref value which is what I think you should be checking here

I think, that if it is needed to add such a pref, then it should be done in separate issue.

Reply:
yea, as mentioned in dm I guess this is needed right now